server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/SearchAction.java

```diff
@@ -56,137 +51,144 @@
   private static final String FIELD_PARENT_NAME = "parentName";
   private static final String FIELD_ACTIVE_RULE_COUNT = "activeRuleCount";
   private static final String FIELD_PROJECT_COUNT = "projectCount";
+  private static final String FIELD_RULES_UPDATED_AT = "rulesUpdatedAt";
 
   private static final Set<String> ALL_FIELDS = ImmutableSet.of(
     FIELD_KEY, FIELD_NAME, FIELD_LANGUAGE, FIELD_LANGUAGE_NAME, FIELD_IS_INHERITED, FIELD_PARENT_KEY, FIELD_PARENT_NAME, FIELD_IS_DEFAULT, FIELD_ACTIVE_RULE_COUNT,
-    FIELD_PROJECT_COUNT);
+    FIELD_PROJECT_COUNT, FIELD_RULES_UPDATED_AT);
 
-  private static final String PARAM_LANGUAGE = FIELD_LANGUAGE;
+  static final String PARAM_LANGUAGE = FIELD_LANGUAGE;
+  static final String PARAM_PROJECT_KEY = "projectKey";
+  static final String PARAM_DEFAULT = "default";
+  static final String PARAM_PROFILE_NAME = "profileName";
 
+  private final SearchDataLoader dataLoader;
   private final Languages languages;
 
-  private final QProfileLookup profileLookup;
-
-  private final QProfileLoader profileLoader;
-
-  private final QualityProfileDao qualityProfileDao;
-
-  public SearchAction(Languages languages, QProfileLookup profileLookup, QProfileLoader profileLoader, QualityProfileDao qualityProfileDao) {
+  public SearchAction(SearchDataLoader dataLoader, Languages languages) {
+    this.dataLoader = dataLoader;
     this.languages = languages;
-    this.profileLookup = profileLookup;
-    this.profileLoader = profileLoader;
-    this.qualityProfileDao = qualityProfileDao;
   }
 
   @Override
   public void define(WebService.NewController controller) {
-    NewAction search = controller.createAction("search")
+    NewAction action = controller.createAction("search")
       .setSince("5.2")
       .setDescription("List quality profiles.")
       .setHandler(this)
+      .addFieldsParam(ALL_FIELDS)
```

Comment:
The field "key" must always be returned. Do you really think that this parameter is needed ?